### PR TITLE
No punctures

### PR DESCRIPTION
**Pull request by llibert94**
I updated CubicHorndeski as we discussed today
In this PR there's as well the new SimulationParametersBase
All tests pass, everything compiles and the Kerr and BinaryBH4dST examples run (I suppose the others as well, but haven't checked it)

### CODE REVIEW

File: Examples/BinaryBHCubicHorndeski/params.txt [hunk 1]
```diff
@@ -149,24 +146,23 @@ lapse_advec_coeff = 1.0
...
 kappa3 = 1.
 covariantZ4 = 1 # 0: keep kappa1; 1 [default]: replace kappa1 -> kappa1/lapse
 
 # Modified CCZ4 parameters
 a0 = 0.2
 b0 = 0.4
 
-# Cubic Horndeski parameters
-g2 = 0.
+# 4dST parameters
```
Comment (KAClough) [inline Examples/BinaryBHCubicHorndeski/params.txt:164]: Think this should be cubic horndeski :-)

File: Examples/KerrBH4dST/CouplingAndPotential.hpp [hunk 2]
```diff
@@ -51,7 +51,7 @@ class CouplingAndPotential
                  exp(-m_params.quadratic_factor * vars.phi * vars.phi *
                      (1. + m_params.quartic_factor * vars.phi * vars.phi)) *
                  vars.phi *
-                 (1. + 2. * m_params.quadratic_factor * vars.phi * vars.phi);
+                 (1. + 2. * m_params.quartic_factor * vars.phi * vars.phi);
```
Comment (KAClough) [inline Examples/KerrBH4dST/CouplingAndPotential.hpp:54]: I guess this was a typo originally?

File: Source/CubicHorndeski/CubicHorndeski.impl.hpp [hunk 3]
```diff
@@ -55,33 +69,49 @@ CubicHorndeski<coupling_and_potential_t>::compute_Sij_TF_and_S(
...
+        dphi_dot_dphi += vars.chi * h_UU[i][j] * d1.phi[i] * d1.phi[j];
+    }
+
+    // X
+    data_t X = 0.5 * (vars.Pi * vars.Pi - dphi_dot_dphi);
+
     // compute potential and add contributions to EM Tensor
     const data_t chi_regularised = simd_max(vars.chi, 1e-6);
 
+    data_t repeated_quantity =
```
Comment (KAClough) [inline Source/CubicHorndeski/CubicHorndeski.impl.hpp:94]: Hmm... why repeated? If we can't think of a good name we could call it "another_useful_quantity", which would be funny (I think so anyway) :-)

File: Source/CubicHorndeski/CubicHorndeski.impl.hpp [hunk 4]
```diff
@@ -148,43 +179,37 @@ void CubicHorndeski<coupling_and_potential_t>::pre_compute_no_gauge(
...
     // dphi_dot_dphi - not conformal
     data_t dphi_dot_dphi = 0.;
-    FOR(i, j) { dphi_dot_dphi += h_UU[i][j] * d1.phi[i] * d1.phi[j]; }
-    dphi_dot_dphi *= chi;
+    FOR(i, j)
+    {
+        dphi_dot_dphi += vars.chi * h_UU[i][j] * d1.phi[i] * d1.phi[j];
+    }
 
     // X
```
Comment (KAClough) [inline Source/CubicHorndeski/CubicHorndeski.impl.hpp:189]: This looks funny as a comment! Maybe expand to say X is the kinetic term in the Lagrangian or something like that.

**Comment by KAClough** [comment]
Look great! Only a few minor comments to address.

File: Examples/BinaryBHCubicHorndeski/params.txt (hunk 1 above)
Comment (llibert94) [reply Examples/BinaryBHCubicHorndeski/params.txt:164]: Oh yes I'll correct it

File: Examples/KerrBH4dST/CouplingAndPotential.hpp (hunk 2 above)
Comment (llibert94) [reply Examples/KerrBH4dST/CouplingAndPotential.hpp:54]: Yes it was! I figured it out when the scalarisation of Kerr was giving me different results

File: Source/CubicHorndeski/CubicHorndeski.impl.hpp (hunk 3 above)
Comment (llibert94) [reply Source/CubicHorndeski/CubicHorndeski.impl.hpp:94]: Okay I'll put it! I didn't come up with any better name hahah

**Review by KAClough — APPROVED** [review]

**Pull request merged**